zerver/actions/streams.py

```diff
@@ -1180,7 +1180,11 @@ def send_change_stream_permission_notification(
             new_policy=new_policy_name,
         )
         internal_send_stream_message(
-            sender, stream, str(Realm.STREAM_EVENTS_NOTIFICATION_TOPIC_NAME), notification_string
+            sender,
+            stream,
+            str(Realm.STREAM_EVENTS_NOTIFICATION_TOPIC_NAME),
+            notification_string,
+            archived_channel_notice=stream.deactivated,
```

Comment:
Another way would be to update `access_stream_for_send_message`:
```diff
    if stream.deactivated:
-       if archived_channel_notice:
+       if topic_name == str(Realm.STREAM_EVENTS_NOTIFICATION_TOPIC_NAME):
            return
        raise JsonableError(
            _("Not authorized to send to channel '{channel_name}'").format(channel_name=stream.name)
        )
```

The downside is string comparison instead of boolean check. Upside is not worrying about setting `archived_channel_notice` correctly.